diff --git a/source/lexbor/core/conv.c b/source/lexbor/core/conv.c
index d8ec21b..299244b 100644
--- a/source/lexbor/core/conv.c
+++ b/source/lexbor/core/conv.c
@@ -75,7 +75,9 @@ lexbor_conv_int64_to_data(int64_t num, lxb_char_t *buf, size_t len)
     }
 
     i = length;
-    buf[length] = '\0';
+    if (length < len) {
+        buf[length] = '\0';
+    }
 
     while (i != have_minus) {
         i -= 1;
