diff --git a/quickjs.c b/quickjs.c
index 1111111..2222222 100644
--- a/quickjs.c
+++ b/quickjs.c
@@ -43784,11 +43784,6 @@ static int js_array_cmp_generic(const void *a, const void *b, void *opaque) {
     if (psc->exception)
         return 0;

     if (psc->has_method) {
-        /* custom sort function is specified as returning 0 for identical
-         * objects: avoid method call overhead.
-         */
-        if (!memcmp(&ap->val, &bp->val, sizeof(ap->val)))
-            goto cmp_same;
         argv[0] = ap->val;
         argv[1] = bp->val;
         res = JS_Call(ctx, psc->method, JS_UNDEFINED, 2, argv);
