diff --git a/meson.build b/meson.build
index a444f10..33843c0 100644
--- a/meson.build
+++ b/meson.build
@@ -132,7 +132,9 @@ qjs_sys_deps = []
 
 m_dep = cc.find_library('m', required: false)
 qjs_sys_deps += m_dep
-qjs_sys_deps += dependency('threads', required: false)
+if host_system != 'windows'
+  qjs_sys_deps += dependency('threads', required: false)
+endif
 qjs_sys_deps += dependency('dl', required: false)
 
 qjs_srcs = files(
@@ -312,6 +314,7 @@ qjsc_exe = executable(
 
   c_args: qjs_c_args,
   dependencies: [qjs_dep, qjs_libc_dep],
+  pie: host_system != 'windows',
   install: true,
 )
 
@@ -338,6 +341,7 @@ qjs_exe = executable(
   dependencies: [qjs_dep, qjs_libc_dep, mimalloc_dep],
   export_dynamic: true,
 
+  pie: host_system != 'windows',
   install: true,
 )
 
